pypdf/_xobj_image_helpers.py

```diff
@@ -195,7 +195,13 @@ def bits2byte(data: bytes, size: Tuple[int, int], bits: int) -> bytes:
         else:
             if img.mode == "1":
                 # Two values ("high" and "low").
-                assert len(lookup) == 2 * nb, len(lookup)
+                expected_count = 2 * nb
+                if len(lookup) != expected_count:
```

Comment:
That seems to be a spill-over from https://github.com/py-pdf/pypdf/pull/2331. Hence I would first merge that PR.